### PR TITLE
default workspace support

### PR DESCRIPTION
**Pull request by drekle**
Currently a users default workspace is stored in a config element on their local machine.  This will push the default workspace up into remote state, allowing the authentication service to query a users default workspace if it has been set and allow internal services to set the default workspace for the first one provisioned by a registering user, removing the requirement that they authenticate with `--default-org` or know what the workspace even is if they provision only one.

### CODE REVIEW

**Review by cyberswat — APPROVED** [review]

File: live/administration/v1alpha1/workspace.proto [hunk 1]
```diff
@@ -169,4 +169,26 @@ message DeleteWorkspaceDeveloperResponse {
...
     */
     Workspace workspace = 1;
+}
+
+message SetDefaultWorkspaceRequest {
+    /*
+    `Required`
+    The id of the subscription for the workspace
+    */
+    string subscription = 1;
```
Comment (wozniakjan) [inline live/administration/v1alpha1/workspace.proto:179]: nit: shouldn't this take some form of `UserID` as well? I can imagine 5 users sharing the same subscription but having a desire for a different default workspace?

**Review by wozniakjan — APPROVED** [review]
this is much needed!

File: live/administration/v1alpha1/workspace.proto (hunk 1 above)
Comment (drekle) [reply live/administration/v1alpha1/workspace.proto:179]: I can gather a bit of information such as this from the Bearer Token.

File: live/administration/v1alpha1/workspace.proto [hunk 2]
```diff
@@ -169,4 +169,26 @@ message DeleteWorkspaceDeveloperResponse {
...
+
+    /*
+    The name of the workspace in the subscription
+    */
+    string workspace = 2;
```
Comment (nic-hima) [inline live/administration/v1alpha1/workspace.proto:184]: is this a required field??

Comment (drekle) [reply live/administration/v1alpha1/workspace.proto:184]: yes, thanks.  Will update for documentation.

**Review by josefkarasek — APPROVED** [review]
lgtm

File: .github/workflows/doc.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: Doc
+on: 
+  push:
+    branches:
+      - main
+jobs:
+  publish:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: release-doc
```
Comment (nic-hima) [inline .github/workflows/doc.yaml:10]: thanks for adding this workflow

File: live/administration/v1alpha1/workspace.proto [hunk 4]
```diff
@@ -50,6 +50,11 @@ message Workspace {
     The ID of the subscription which this workspace belongs
     */
     string subscription = 4;
+
+    /*
+    Optional metadata information about this workspace
+    */
+    map<string, string> metadata = 16;
```
Comment (nic-hima) [inline live/administration/v1alpha1/workspace.proto:57]: what is the intended usage of this metadata map?

**Review by nic-hima — APPROVED** [review]
lgtm 👍

File: live/administration/v1alpha1/workspace.proto (hunk 4 above)
Comment (drekle) [reply live/administration/v1alpha1/workspace.proto:57]: Hopefully to provide stop-gap solutions for the CLI in the short term.  Currently the CLI uses namespace names, however moving away from that we will temporarily need to query namespaces from a source we can remove later without modifying the protobuf spec.  Users may want to pass in a subscriptionID.Workspace qualified workspace name and retrieve the namespace name for deprecated API's only.  Workspace.Name will contain a display name with now capability of getting at the generated namespace name.

_We may also allow users to set their own metadata and query resources with metadata filters however that is not the reason for the addition and I struggle to see that being useful on workspaces, perhaps sites._

Comment (nic-hima) [reply live/administration/v1alpha1/workspace.proto:57]: > We may also allow users to set their own metadata and query resources with metadata filters however that is not the reason for the addition and I struggle to see that being useful on workspaces, perhaps sites.

this is a re-implementation of the `tag` functionality, but for Workspaces??

**Pull request merged**